modules/tinymce/src/themes/silver/main/ts/ContextToolbar.ts

```diff
@@ -52,11 +53,27 @@ const anchorOverrides = {
 };
 
 // On desktop we prioritise north-then-south because it's cleaner, but on mobile we prioritise south to try to avoid overlapping with native context toolbars
-const desktopAnchorSpecLayouts = {
-  onLtr: () => [ Layout.north, Layout.south, Layout.northeast, Layout.southeast, Layout.northwest, Layout.southwest,
-    LayoutInside.north, LayoutInside.south, LayoutInside.northeast, LayoutInside.southeast, LayoutInside.northwest, LayoutInside.southwest ],
-  onRtl: () => [ Layout.north, Layout.south, Layout.northwest, Layout.southwest, Layout.northeast, Layout.southeast,
-    LayoutInside.north, LayoutInside.south, LayoutInside.northwest, LayoutInside.southwest, LayoutInside.northeast, LayoutInside.southeast ]
+
+const getDesktopAnchorSpecLayouts = (contextbar: AlloyComponent) => {
+  // We try to keep our current pinned layout, so we identify any classes that are on the contextbar,
+  // and prefer that pinned layout to the other pinned layout
+  const getPinnedLayouts = () => {
+    // Something at the top will be anchored from its bottom value and grow upward
+    const isAlreadyAtTop = isElementTopAligned(contextbar.element);
+
+    return isAlreadyAtTop ? [ PinnedLayout.pinAtTop, PinnedLayout.pinAtBottom ] : [ PinnedLayout.pinAtBottom, PinnedLayout.pinAtTop ];
```

Comment:
Maybe rather than passing `isAlreadyAtTop` through to silver, we could just make a `PinnedLayout.auto` function that takes an element as an argument and returns the correct tuple based on which way it's already aligned?

Reply:
I have some hesitation about that solution, since then alloy would decide which is the 'right' order, when all the other orders are decided elsewhere, like in this file. Even if it's unlikely to change.